apps/wallet/src/ui/app/hooks/useRecentTransactions.ts

```diff
@@ -224,3 +221,30 @@ export function useRecentTransactions() {
         }
     );
 }
+
+export function useRecentTransaction(transactionDigest: string | null) {
+    const rpc = useRpc();
+    const userAddress = useAppSelector((state) => state.account.address);
+    return useQuery({
+        queryKey: [
+            'recent transaction',
+            'get transaction with effects',
+            transactionDigest,
+        ],
+        queryFn: async () => {
+            const transactionResponse = await rpc.getTransactionWithEffects(
+                transactionDigest!
+            );
+            const processedTransactions = await processTransactionEffects(
+                [transactionResponse],
+                userAddress!,
+                rpc
+            );
+            return processedTransactions?.[0] || null;
+        },
+        enabled: !!transactionDigest?.length && !!userAddress,
+        staleTime: 10 * 1000,
+        retry: 10,
+        retryDelay: (attemptIndex) => 1000 + 1000 * attemptIndex,
```

Comment:
Curious why we didn't just use the default one? I assume the delay got too high?

Reply:
I think the default one is 1 second, right?